buffer/src/main/java/io/netty/buffer/ReadOnlyUnsafeDirectByteBuf.java

```diff
@@ -27,8 +27,10 @@
 final class ReadOnlyUnsafeDirectByteBuf extends ReadOnlyByteBufferBuf {
     private final long memoryAddress;
 
-    ReadOnlyUnsafeDirectByteBuf(ByteBufAllocator allocator, ByteBuffer buffer) {
-        super(allocator, buffer);
+    ReadOnlyUnsafeDirectByteBuf(ByteBufAllocator allocator, ByteBuffer byteBuffer) {
+        super(allocator, byteBuffer);
+        // Use buffer as the super class will slice the passed in ByteBuffer which means the memoryAddress
+        // may be different if the position != 0.
```

Comment:
I must be blind, what changed here?

Reply:
I had to do a double take when I was reviewing as well. `super()` uses the `ByteBuffer` but the line below uses the `buffer` from the parent class.

Reply:
@carl-mastrangelo as @Scottmitch said... we now use the `ByteBuffer` stored in super. Before we used the one that was passed into the constructor.